engine/gameobject/src/gameobject/gameobject.cpp

```diff
@@ -969,8 +969,8 @@ namespace dmGameObject
     dmhash_t ConstructInstanceId(uint32_t index)
     {
         char buffer[16] = { 0 };
-        dmSnPrintf(buffer, sizeof(buffer), "%sinstance%d", ID_SEPARATOR, index);
-        return dmHashString64(buffer);
+        int length = dmSnPrintf(buffer, sizeof(buffer), "%sinstance%d", ID_SEPARATOR, index);
+        return dmHashBuffer64(buffer, length);
```

Comment:
No compiler warnings here?

Reply:
I couldn't find any warning. But that might be a Windows thing. In any case, I added a cast to `uint32_t`.